### PR TITLE
add support for latest version of code-server

### PR DESCRIPTION
**Pull request by bpmct**
## Summary

Probably makes sense to merge #2223 first. This fixes some the code-server guide to use the latest version of code-server:
- `allow-http` flag does not exist
- `--disable-auth` is now `--auth none`
- change to port 8080 (new default code-server port)
- open `/home/coder/project` by default in code-server
- permissions fix?
- add note about container development and dependencies
- fix missing `cd` in pomerium build example

## Checklist

- [x] reference any related issues
- [x] updated docs
- [ ] updated unit tests
- [ ] updated UPGRADING.md
- [x] add appropriate tag (`improvement` / `bug` / etc)
- [x] ready for review

edit: working local docker-compose project here: https://github.com/bpmct/pomerium-code-server

### CODE REVIEW

File: docs/guides/code-server.md [hunk 1]
```diff
@@ -64,15 +64,13 @@ codeserver:
...
-    - 8443:8443
+    - 8080:8080
   volumes:
     - ./code-server:/home/coder/project
-  command: --allow-http --no-auth --disable-telemetry
+  command: --auth none --disable-telemetry /home/coder/project
 ```
 
-Note we are mounting a directory called`./code-server`. Be sure to give the default docker user write permissions to that folder by running `chown -R 1000:1000 code-server/`.
-
```
Comment (bpmct) [inline docs/guides/code-server.md:74]: When applied, this ended up making the filesystem unwritable for me. Worked well without.

I may be missing something, but I know our Dockerfile now [fixes a debian uid error](https://github.com/cdr/code-server/blob/main/ci/release-image/Dockerfile#L28-L33).

File: docs/guides/code-server.md [hunk 2]
```diff
@@ -107,21 +109,26 @@ Finally, now that we've got Go all we need to go is grab the latest source and b
...
+git clone https://github.com/pomerium/pomerium.git
+
 # build pomerium
-$ make build
+cd pomerium && make build
 # run pomerium!
-$ ./bin/pomerium --version
-v0.2.0+e1c00b1
+./bin/pomerium --version
+# v0.2.0+e1c00b1
```
Comment (bpmct) [inline docs/guides/code-server.md:118]: FYI this failed for me inside code-server. I was planning on updating the version number

```sh
coder@929f14a68fd0:~/project/pomerium$ make build
==> build
./scripts/get-envoy.bash
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   625  100   625    0     0   3033      0 --:--:-- --:--:-- --:--:--  3033
100 58.9M  100 58.9M    0     0  7347k      0  0:00:08  0:00:08 --:--:-- 8291k
build github.com/pomerium/pomerium/cmd/pomerium: cannot load crypto/ed25519: cannot find module providing package crypto/ed25519
make: *** [Makefile:90: build] Error 1
```

Comment (bpmct) [reply docs/guides/code-server.md:118]: (the other packages were installed, this was the second time running to get the error again)

File: docs/guides/code-server.md [hunk 3]
```diff
@@ -85,6 +83,10 @@ As a final touch, now that we've done all this work we might as well use our new
 To build Pomerium, we must [install go](https://golang.org/doc/install) which is as simple as running the following commands in the [integrated terminal].
 
 ```bash
+# install dependencies with apt
+sudo apt-get update && sudo apt-get install -y wget make
+
+# download go
 wget https://dl.google.com/go/go1.12.7.linux-amd64.tar.gz
```
Comment (desimone) [inline docs/guides/code-server.md:90]:
```suggestion
wget https://golang.org/dl/go1.16.4.linux-amd64.tar.gz
```

File: docs/guides/code-server.md [hunk 4]
```diff
@@ -85,6 +83,10 @@ As a final touch, now that we've done all this work we might as well use our new
...
 sudo tar -C /usr/local -xzf go1.12.7.linux-amd64.tar.gz
```
Comment (desimone) [inline docs/guides/code-server.md:91]:
```suggestion
sudo tar -C /usr/local -xzf go1.16.4.linux-amd64.tar.gz
```

File: docs/guides/code-server.md (hunk 2 above)
Comment (desimone) [reply docs/guides/code-server.md:118]: I added some suggestions. I think (hope?) that the updated version of Go should solve that issue.

File: docs/guides/code-server.md [hunk 5]
```diff
@@ -107,21 +109,26 @@ Finally, now that we've got Go all we need to go is grab the latest source and b
...
-v0.2.0+e1c00b1
+./bin/pomerium --version
+# v0.2.0+e1c00b1
 ```
 
 Happy remote hacking!!!😁
 
+:::tip
+
+When the code-server container is rebuilt, any files outside of `/home/coder/project` are reset, removing any dependencies (such as go and make). In a real remote development workflow, you could mount additional volumes, or [use a custom code-server container](https://github.com/cdr/deploy-code-server/tree/main/deploy-container) with these dependencies installed.
```
Comment (desimone) [inline docs/guides/code-server.md:125]: Good catch -- thank you

**Review by desimone — COMMENTED** [review]
Thanks for the contribution @bpmct! 

LGTM, I just added one suggestion that I hope will fix the pomerium build by updating to the latest version of go.

**Comment by bpmct** [comment]
@desimone that did the trick!

**Review by desimone — APPROVED** [review]
LGTM pending merge conflict resolution. Thanks!

**Pull request merged**